operator/pkg/kube/operator/operator.go

```diff
@@ -0,0 +1,240 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package operator
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"istio.io/istio/pkg/kube"
+	"istio.io/istio/pkg/log"
+	"github.com/ghodss/yaml"
+	corev1 "k8s.io/api/core/v1"
+	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+        apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+        "k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clientcmd"
+        apiclient "k8s.io/client-go/kubernetes"
+        "k8s.io/client-go/rest"
+	"k8s.io/client-go/util/workqueue"
+)
+
+const (
+	operatorLabel = "istio-operator"
+	maxRetries = 5
+)
+
+var (
+        runtimeScheme = runtime.NewScheme()
+        codecs        = serializer.NewCodecFactory(runtimeScheme)
+        deserializer  = codecs.UniversalDeserializer()
+)
+
+// LoadKubeConfig is a unit test override variable for loading the k8s config.
+// DO NOT USE - TEST ONLY.
+var LoadKubeConfig = clientcmd.Load
+
+// CreateInterfaceFromClusterConfig is a unit test override variable for interface create.
+// DO NOT USE - TEST ONLY.
+var CreateInterfaceFromClusterConfig = kube.CreateInterfaceFromClusterConfig
+
+// addConfigMapCallback prototype for unit testing.
+// DO NOT USE - TEST ONLY.
+type addConfigMapCallback func(*corev1.ConfigMap) error
+
+// removeConfigMapCallback prototype for unit testing.
+// DO NOT USE - TEST ONLY.
+type removeConfigMapCallback func(string) error
+
+// addConfigMapCallback is called during unit tetsts to verify this controller is working properly.
+// DO NOT USE - TEST ONLY.
+//var addConfigMapCallback = nil
+
+// rmoveConfigMapCallback is called during unit tetsts to verify this controller is working properly.
+// DO NOT USE - TEST ONLY.
+//var removeConfigMapCallback = nil
+
+// Controller is the controller implementation for ConfigMap resources.
+type Controller struct {
+        clientExt *apiextensionsclient.Clientset
+	kuberest  *rest.Config
+	namespace string
+	queue     workqueue.RateLimitingInterface
+	informer  cache.SharedIndexInformer
+}
+
+func init() {
```

Comment:
imho you do not need to use `init()` here..

Reply:
agreed.